x/adminmodule/types/message_submit_proposal_legacy.go

```diff
@@ -0,0 +1,98 @@
+package types
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
+	govcodec "github.com/cosmos/cosmos-sdk/x/gov/codec"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	govtypesv1b1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1beta1"
+	"github.com/cosmos/gogoproto/proto"
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	_ sdk.Msg                          = &MsgSubmitProposalLegacy{}
+	_ cdctypes.UnpackInterfacesMessage = &MsgSubmitProposalLegacy{}
+)
+
+func NewMsgSubmitProposalLegacy(content govtypesv1b1.Content, proposer sdk.AccAddress) (*MsgSubmitProposalLegacy, error) {
+	m := &MsgSubmitProposalLegacy{
+		Proposer: proposer.String(),
+	}
+
+	err := m.SetContent(content)
+	if err != nil {
+		return nil, err
+	}
+	return m, nil
+}
+
+func (m *MsgSubmitProposalLegacy) GetContent() govtypesv1b1.Content {
+	content, ok := m.Content.GetCachedValue().(govtypesv1b1.Content)
+	if !ok {
+		return nil
+	}
+	return content
+}
+
+func (m *MsgSubmitProposalLegacy) SetContent(content govtypesv1b1.Content) error {
+	msg, ok := content.(proto.Message)
+	if !ok {
+		return fmt.Errorf("can't proto marshal %T", msg)
```

Comment:
can't proto marshal? I think it would be better to phrase it like "failed to cast proposal content of type %T to proto.Message" or something

Reply:
This is the way how it's done in current admin module & `x/gov `in sdk. So there his no point to rewrite _legacy_ code